roles/ceph-common/tasks/main.yml

```diff
@@ -0,0 +1,88 @@
+---
+- include: check_system_and_vars.yml
+
+- name: install dependencies
+  apt: pkg={{ item }}
+       state=present
+       update_cache=yes
+       cache_valid_time=3600
+  with_items: ceph_common.debian_pkgs
+
+- name: install the ceph repository stable key
+  apt_key: url={{ ceph_common.ceph_stable_key }}
+           state=present
+  when: ceph_common.ceph_stable
+
+- name: add ceph stable repository
+  apt_repository: repo="deb http://ceph.com/debian-{{ ceph_common.ceph_stable_release }}/ {{ ansible_lsb.codename }} main"
+                  state=present
+  when: ceph_common.ceph_stable
+
+- name: install ceph
+  apt: pkg={{ item }}
+       state=present
+  with_items: ceph_common.ceph_pkgs
+
+- name: configure rbd clients directories
+  file: path={{ item }}
+        state=directory
+        owner=libvirt-qemu
+        group=kvm
+        mode=0755
+  with_items: ceph_common.rbd_client_paths
+  when: ceph_common.rbd_client_directories
+
+- name: check for a ceph socket
+  stat: path=/var/run/ceph/*.asok
+  changed_when: false
+  failed_when: false
+  register: socket
+
+# we're storing the fsid in a file on the first monitor node
+- name: check if fsid file exists
+  stat: path=/etc/ceph/fsid
+  run_once: true
+  delegate_to: "{{ groups['mons'][0] }}"
+  failed_when: false
+  register: fsid_file_check
+
+- name: generate fsid file if we don't have one
+  shell: uuidgen > /etc/ceph/fsid
+  run_once: true
+  delegate_to:  "{{ groups['mons'][0] }}"
+  when: fsid_file_check.stat.exists == false
+
+- name: fetch contents of fsid file
+  slurp: path=/etc/ceph/fsid
+  run_once: true
+  delegate_to: "{{ groups['mons'][0] }}"
+  register: fsid_file
+
+# for redundancy, and if the hosts in mons get reordered
+- name: copy fsid to all hosts
+  copy:
+    dest: "{{ hostvars[groups['mons'][0]]['fsid_file']['source'] }}"
+    content: "{{ hostvars[groups['mons'][0]]['fsid_file']['content'] | b64decode }}"
```

Comment:
Have you tried just calling:
`"{{ fsid_file['source'] }}"`
and
`"{{ fsid_file['content'] }}"` ?

Because you use run_once when registering fsid_file, ever host in the play will get that variable. It's a quirk of run_once, but it'll make this task easier to read.